### PR TITLE
[spec] add Gather-8 with negative indices

### PR DESCRIPTION
**Pull request by pavel-esir**
### Details:
 - the only difference from Gather-7 is that negative `indices` are allowed
 - required nGraph transformations to support old behavior for negative indices is under consideration
 
### Tickets:
 - 54622

### CODE REVIEW

File: docs/ops/movement/Gather_8.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
+## Gather <a name="Gather"></a> {#openvino_docs_ops_movement_Gather_7}
+
+**Versioned name**: *Gather-7*
```
Comment (popovaan) [inline docs/ops/movement/Gather_8.md:3]: Gather-7 -> Gather-8?
openvino_docs_ops_movement_Gather_7 -> openvino_docs_ops_movement_Gather_8

Comment (pavel-esir) [reply docs/ops/movement/Gather_8.md]: thanks! corrected

**Review by evolosen — APPROVED** [review]

**Review by popovaan — APPROVED** [review]

File: docs/ops/movement/Gather_8.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
+## Gather <a name="Gather"></a> {#openvino_docs_ops_movement_Gather_8}
```
Comment (lazarevevgeny) [inline docs/ops/movement/Gather_8.md:1]: Update opset_8.md

Comment (pavel-esir) [reply docs/ops/movement/Gather_8.md:1]: done

**Review by achetver — APPROVED** [review]

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**